File: 7.Search-Algorithm/Example/Binary/Binary.java
public class BinarySearch {
    public static int binarySearch(int[] arr, int key) {
        int start = 0;
        int end = arr.length - 1;
        while (start <= end) {
            int mid = (start + end) / 2;
            if (arr[mid] == key) {
                return mid;
            }
            if (key < arr[mid]) {
                end = mid - 1;
            } else {
                start = mid + 1;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        int[] arr = {0, 1, 2, 3, 4, 6, 100, 10000};
        System.out.println(binarySearch(arr, 100)); // 6
    }
}

File: 7.Search-Algorithm/Example/Binary/Binary.js
function binarySearch(arr, key) {
  let start = 0,
    end = arr.length - 1;
  while (start <= end) {
    let mid = Math.floor((start + end) / 2);
    if (arr[mid] === key) {
      return mid;
    }
    if (key < arr[mid]) {
      end = mid - 1;
    } else {
      start = mid + 1;
    }
  }
  return -1;
}
const arr = [0,1,2,3,4,6,100,10000];

console.log(binarySearch(arr, 100)); // 6